### PR TITLE
[cmake] Link Kokkos::kokkos

### PR DESCRIPTION
**Pull request by vgvassilev**

### CODE REVIEW

File: unittests/Kokkos/CMakeLists.txt [hunk 1]
```diff
@@ -12,5 +12,6 @@ if (NOT (LLVM_REQUIRES_RTTI OR LLVM_ENABLE_RTTI))
   endif()
 endif()
 
+target_link_libraries(KokkosTests PRIVATE Kokkos::kokkos)
```
Comment (vgvassilev) [inline unittests/Kokkos/CMakeLists.txt:15]: Hi @kliegeois, that does not play very well with the Kokkos distribution on ubuntu that we use for out bots. I've noticed that and that's why I added the line after this one. Does it not work for you as it was?

Comment (kliegeois) [reply unittests/Kokkos/CMakeLists.txt:15]: Hi @vgvassilev ! Do you know the version number of the Kokkos you used? Without the line of this commit, I was unable to use the latest Kokkos on my Mac for the unit tests. Maybe this is due to a difference in Kokkos version.

Comment (vgvassilev) [reply unittests/Kokkos/CMakeLists.txt:15]: The Kokkos version that we install is `libtrilinos-kokkos-dev` for ubuntu 22.04 which seems to install `libtrilinos-kokkos-13.2`.

Comment (vgvassilev) [reply unittests/Kokkos/CMakeLists.txt:15]: @kliegeois, ping.

**Comment by vgvassilev** [comment]
@gojakuch, you mentioned you have a similar problem where you needed to add locally `Kokkos::kokkos`. Was your build from source?

**Comment by vgvassilev** [comment]
Looks like this is not needed for our bots. Let’s revisit if there is a need to.

**Pull request closed without merging**